source/tutorials/2024-adding-bioinformatic-software-to-bioconda.rst

```diff
@@ -0,0 +1,460 @@
+*You can see the original post* `here <https://ubinfie.github.io/2024/08/16/adding-to-bioconda-quickguide.html>`_
+
+This tutorial aims to give a gentle introduction to adding bioinformatic software to Bioconda.
+
+The main sections of this tutorial are:
+
+- TL;DR
+- Prerequisites
+- Adding a new tool or package
+- Debugging a recipe build
+- Updating an existing tool or package recipe
+
+Please note that this tutorial comes with 'no warranty'!(!), as the Bioconda build steps could change at any point.
+However, the steps here should act as a good starting point.
+Furthermore, if you are planning to add someone else's tool or package to Bioconda, it's always good etiquette to ask or just inform the original authors that it will happen.
+
+TL;DR Relevant Commands
+*****************
+
+.. code-block:: bash
+
+    ## Create environment with conda building tools
+    conda create -n bioconda-build -c conda-forge -c bioconda conda-build bioconda-utils greyskull
+    conda activate bioconda-build
+
+    ## Clone repo of your fork of https://github.com/bioconda/bioconda-recipes and make branch
+    git clone <your-forked-bioconda-recipes-repo-address>
+    git switch -c add-<toolname>
+
+    ## Make recipe meta.yaml
+    ## Option 1: If using Greyskull
+    cd recipes/
+    greyskull <pypi/cran> > <toolname>/meta.yaml
+
+    ## Option 2: If not using Greyskull
+    mkdir recipes/<toolname>
+    touch recipes/<toolname>/meta.yaml
+
+    ## Lint recipe meta.yaml
+    bioconda-utils lint recipes/ --packages pathphynder
+
+    ## Perform a local build (two options)
+    ## Option 1:
+    conda build recipes/<toolname>
+
+    ## Option 2:
+    bioconda-utils build --docker --mulled-test --packages <toolname>
+
+    ## Debugging
+    ## Option 1: conda-build
+    cd /<path>/<to>/<conda-install>/envs/<toolname>/conda-bld/linux-64
+    conda create -n <debugging-env-name> -c ./ <tool_name_as_in_recipe>
+    conda build recipes/<toolname> --keep-old-work
+
+    ## Option 2: bioconda-utils
+    docker run -t --net host --rm -v /tmp/tmp<randomletters>/build_script.bash:/opt/build_script.bash -v /<path>/<to>/<conda-install>/envs/<toolname>/conda-bld/:/opt/host-conda-bld -v /<path>/<to>/<recipes_local_clone>/recipes/<toolname>:/opt/recipe -e LC_ADDRESS=en_GB.UTF-8 -e LC_NAME=en_GB.UTF-8 -e LC_MONETARY=en_GB.UTF-8 -e LC_PAPER=en_GB.UTF-8 -e LANG=en_GB.UTF-8 -e LC_IDENTIFICATION=en_GB.UTF-8 -e LC_TELEPHONE=en_GB.UTF-8 -e LC_MEASUREMENT=en_GB.UTF-8 -e LC_TIME=en_GB.UTF-8 -e LC_NUMERIC=en_GB.UTF-8 -e HOST_USER_ID=1000 quay.io/bioconda/bioconda-utils-build-env-cos7:2.11.1 bash
+
+    conda mambabuild -c file:///opt/host-conda-bld --override-channels --no-anaconda-upload -c conda-forge -c bioconda -c defaults -e /opt/host-conda-bld/conda_build_config_0_-e_conda_build_config.yaml -e /opt/host-conda-bld/conda_build_config_1_-e_bioconda_utils-conda_build_config.yaml /opt/recipe/meta.yaml 2>&1
```

Comment:
do we still need mambabuild or can we just use conda build?

Reply:
I don't know, I took this command from what the Azure bioconda runners were executing at the time I made my own notes, has this been changed?